searx/static/themes/simple/src/js/main/search.js

```diff
@@ -35,118 +32,105 @@
+      formData.append("q", query);
+      request = fetch("./autocompleter", {
+        method: 'POST',
+        body: formData,
+      });
+    }
+
+    request.then(async function (response) {
+      const results = await response.json();
+
+      if (!results) return;
+
+      const autocomplete = d.querySelector(".autocomplete");
+      const autocompleteList = d.querySelector(".autocomplete ul");
+      autocomplete.classList.add("open");
+      autocompleteList.innerHTML = "";
+
+      // show an error message that no result was found
+      if (!results[1] || results[1].length == 0) {
+        const noItemFoundMessage = document.createElement("li");
+        noItemFoundMessage.classList.add('no-item-found');
+        noItemFoundMessage.innerHTML = searxng.settings.translations.no_item_found;
+        autocompleteList.appendChild(noItemFoundMessage);
+        return;
+      }
+
+      for (let result of results[1]) {
+        const li = document.createElement("li");
+        li.innerText = result;
+
+        searxng.on(li, 'click', () => {
+          qinput.value = result;
+          const form = d.querySelector("#search");
+          form.submit();
+          autocomplete.classList.remove('open');
+        })
+        autocompleteList.appendChild(li);
+      }
+    });
+  }
+
   searxng.ready(function () {
     qinput = d.getElementById(qinput_id);
+    const autocomplete = d.querySelector(".autocomplete");
+    const autocompleteList = d.querySelector(".autocomplete ul");
 
     if (qinput !== null) {
       // clear button
       createClearButton(qinput);
 
       // autocompleter
       if (searxng.settings.autocomplete) {
-        searxng.autocomplete = AutoComplete.call(w, {
-          Url: "./autocompleter",
-          EmptyMessage: searxng.settings.translations.no_item_found,
-          HttpMethod: searxng.settings.method,
-          HttpHeaders: {
-            "Content-type": "application/x-www-form-urlencoded",
-            "X-Requested-With": "XMLHttpRequest"
-          },
-          MinChars: searxng.settings.autocomplete_min,
-          Delay: 300,
-          _Position: function () {},
-          _Open: function () {
-            var params = this;
-            Array.prototype.forEach.call(this.DOMResults.getElementsByTagName("li"), function (li) {
-              if (li.getAttribute("class") != "locked") {
-                li.onmousedown = function () {
-                  params._Select(li);
-                };
-              }
-            });
-          },
-          _Select: function (item) {
-            AutoComplete.defaults._Select.call(this, item);
-            var form = item.closest('form');
-            if (form) {
-              form.submit();
+        searxng.on(qinput, 'input', () => {
+          const query = qinput.value;
+          if (query.length < searxng.settings.autocomplete_min) return;
+
+          setTimeout(() => {
+            if (query == qinput.value) fetchResults(query);
+          }, 300);
```

Comment:
If you don't mind me asking, how come you are adding a delay to the autocomplete? It seems to make the user experience of using autocomplete much worse.

With autocomplete.js, there was a bug where if the previous request hadn't finished, the next request would cancel and never get sent (so a delay of 300ms was added which would make the issue less apparent).

Let me know your thoughts.

Reply:
> If you don't mind me asking, how come you are adding a delay to the autocomplete? It seems to make the user experience of using autocomplete much worse.
>
The goal of this PR is to rebuild the autocomplete behavior of autocomplete.js without any changes, thus I kept the delay.

> With autocomplete.js, there was a bug where if the previous request hadn't finished, the next request would cancel and never get sent (so a delay of 300ms was added which would make the issue less apparent).
>
Interesting, I wasn't aware of that.

Nevertheless I think such behavior changes should be discussed in an other issue, as I said the only goal of this PR is to remove the very outdated, unmaintained `autocomplete.js` dependency without changing any behavior.